### DISCUSSION TITLE
About USB FIFO settings

### DISCUSSION THREAD

**Discussion by arilink-tech** [hathach/tinyusb#999] [Q&A]
Previously, when using other USB libraries, I found that they all allocated the USB FIFO during USB initialization, but no relevant content was found on TinyUSB. Is this automatically allocated inside TinyUSB?
like this:
  /* USB_OTG_HS init function */
  MX_USB_OTG_FS_PCD_Init();
  /* USER CODE BEGIN USB_Device_Init_PreTreatment_1 */
  /* Set Rx FIFO */
  HAL_PCDEx_SetRxFiFo(&hpcd_USB_OTG_FS, 0x200);
  /* Set Tx FIFO 0 */
  HAL_PCDEx_SetTxFiFo(&hpcd_USB_OTG_FS, 0, 0x10);
  /* Set Tx FIFO 2 */
  HAL_PCDEx_SetTxFiFo(&hpcd_USB_OTG_FS, 2, 0x10);
  /* Set Tx FIFO 3 */
  HAL_PCDEx_SetTxFiFo(&hpcd_USB_OTG_FS, 3, 0x80);
  /* Set Tx FIFO 4 */
  HAL_PCDEx_SetTxFiFo(&hpcd_USB_OTG_FS, 4, 0x20);

**Comment by hathach** [comment]
yes
